src/main/java/chess/domain/position/Movement.java

```diff
@@ -0,0 +1,74 @@
+package chess.domain.position;
+
+public class Movement {
+
+    private final Position current;
+    private final Position destination;
+
+    public Movement(final Position current, final Position destination) {
+        this.current = current;
+        this.destination = destination;
+    }
+
+    public boolean isVertical() {
+        return current.isSameFile(destination) && !current.isSameRank(destination);
+    }
+
+    public boolean isHorizontal() {
+        return !current.isSameFile(destination) && current.isSameRank(destination);
+    }
+
+    public boolean isDiagonal() {
+        return current.getFileDistance(destination) == current.getRankDistance(destination);
+    }
+
+    public boolean isDiagonalRight() {
+        return isDiagonal()
+                && (!isRight() && !isUp() || isRight() && isUp());
+    }
+
+    public boolean isDiagonalLeft() {
+        return isDiagonal()
+                && (isRight() && !isUp() || !isRight() && isUp());
+    }
+
+    public boolean isUp() {
+        return destination.getRank().isUp(current.getRank());
```

Comment:
getter를 사용하지 않는 좋은 방법은 해당 객체에게 책임을 위임하는 것이죠
```suggestion
        return destination.isUp(current);
```